### PR TITLE
[Config] show proposals when unsupported option is provided

### PR DESCRIPTION
**Pull request by fmata**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28075 
| License       | MIT
| Doc PR        | none

In case of proposals match with the provided option, all proposals are displayed.
In case of no pertinent proposal is available, all options are displayed in alpha order.

### CODE REVIEW

**Review by sstok — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Config/Definition/ArrayNode.php [hunk 1]
```diff
@@ -307,7 +307,30 @@ protected function normalizeValue($value)
...
+                        $minScore = $distance;
+                    }
+                }
+            }
+
             $msg = sprintf('Unrecognized option%s "%s" under "%s"', 1 === \count($value) ? '' : 's', implode(', ', array_keys($value)), $this->getPath());
+
+            if (\count($guesses)) {
+                asort($guesses);
+                $msg .= sprintf('. Did you mean "%s" ?', implode('", "', array_keys($guesses)));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Definition/ArrayNode.php]: the space before `?` should be removed

**Review by nicolas-grekas — APPROVED** [review]
(with minor comment)

File: src/Symfony/Component/Config/Definition/ArrayNode.php (hunk 1 above)
Comment (fmata) [reply src/Symfony/Component/Config/Definition/ArrayNode.php]: Removed.

**Comment by nicolas-grekas** [comment]
Thank you @fmata.

**Pull request merged**